### PR TITLE
Add "exoscale_affinity" data source

### PR DESCRIPTION
**Pull request by falzm**
This change introduces a new `exoscale_affinity` data source.

### CODE REVIEW

File: exoscale/datasource_exoscale_affinity.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+func dataSourceAffinityRead(d *schema.ResourceData, meta interface{}) error {
+	var err error
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutCreate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	req := egoscale.ListAffinityGroups{Type: "host anti-affinity"}
```
Comment (pierre-emmanuelJ) [inline exoscale/datasource_exoscale_affinity.go:41]: I think you can use [listAntiAffinityGroups](https://github.com/exoscale/egoscale/blob/master/antiaffinity_groups.go#L72) call ;)

**Review by pierre-emmanuelJ — APPROVED** [review]
LGTM 👍, see my comment below :)

File: exoscale/datasource_exoscale_affinity.go (hunk 1 above)
Comment (falzm) [reply exoscale/datasource_exoscale_affinity.go:41]: Good catch, I always forget that we have that one now 😉

**Comment by falzm** [comment]
```console
$ make TF_ACC=1 test TESTARGS="-run ^TestAccDataSourceAffinity" 
==> Checking that code complies with gofmt requirements...
go test -v -count=1 -failfast -timeout 120m -run ^TestAccDataSourceAffinity ./...
?   	github.com/terraform-providers/terraform-provider-exoscale	[no test files]
=== RUN   TestAccDataSourceAffinity
--- PASS: TestAccDataSourceAffinity (8.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-exoscale/exoscale	8.547s
?   	github.com/terraform-providers/terraform-provider-exoscale/version	[no test files]
```

**Review by jessicatoscani — APPROVED** [review]

**Pull request merged**